Sharding-Proxing change database error

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/incubator-shardingsphere-example/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response **more than 7 days** and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
4.0.0-RC2
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
Sharding-Proxy
### Expected behavior
No Exception
### Actual behavior
use cloud; #cloud is a database name
### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
Here is my configuration:
```

schemaName: cloud

dataSources:
  cloud_0:
    url: jdbc:mysql://192.168.8.252:8806/cloud?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
    username: root
    password: 123
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 500
  cloud_1:
    url: jdbc:mysql://192.168.8.252:8807/cloud?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
    username: root
    password: 123
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 500

shardingRule:
  defaultDataSourceName: cloud_0
  defaultDatabaseStrategy:
    inline:
      shardingColumn: id
      algorithmExpression: cloud_${id % 2}
  defaultTableStrategy:
    none:
  defaultKeyGenerator:
    type: SNOWFLAKE
    column: id
```
And here is the log:
```
[INFO ] 16:07:04.032 [ShardingSphere-Command-2] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 16:07:04.032 [ShardingSphere-Command-2] ShardingSphere-SQL - Logic SQL: SET PROFILING = 1
[INFO ] 16:07:04.032 [ShardingSphere-Command-2] ShardingSphere-SQL - SQLStatement: DALStatement(super=AbstractSQLStatement(type=DAL, sqlSegments=[], tables=Tables(tables=[]), shardingConditions=Conditions(orConditions=[]), encryptConditions=Conditions(orConditions=[]), parametersIndex=0, logicSQL=SET PROFILING = 1))
[INFO ] 16:07:04.032 [ShardingSphere-Command-2] ShardingSphere-SQL - Actual SQL: cloud_0 ::: SET PROFILING = 1
[INFO ] 16:07:04.034 [ShardingSphere-Command-1] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 16:07:04.034 [ShardingSphere-Command-1] ShardingSphere-SQL - Logic SQL: SHOW STATUS
[INFO ] 16:07:04.034 [ShardingSphere-Command-1] ShardingSphere-SQL - SQLStatement: DALStatement(super=AbstractSQLStatement(type=DAL, sqlSegments=[], tables=Tables(tables=[]), shardingConditions=Conditions(orConditions=[]), encryptConditions=Conditions(orConditions=[]), parametersIndex=0, logicSQL=SHOW STATUS))
[INFO ] 16:07:04.034 [ShardingSphere-Command-1] ShardingSphere-SQL - Actual SQL: cloud_0 ::: SHOW STATUS
[INFO ] 16:07:04.052 [ShardingSphere-Command-4] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 16:07:04.052 [ShardingSphere-Command-4] ShardingSphere-SQL - Logic SQL: SHOW STATUS
[INFO ] 16:07:04.052 [ShardingSphere-Command-4] ShardingSphere-SQL - SQLStatement: DALStatement(super=AbstractSQLStatement(type=DAL, sqlSegments=[], tables=Tables(tables=[]), shardingConditions=Conditions(orConditions=[]), encryptConditions=Conditions(orConditions=[]), parametersIndex=0, logicSQL=SHOW STATUS))
[INFO ] 16:07:04.052 [ShardingSphere-Command-4] ShardingSphere-SQL - Actual SQL: cloud_0 ::: SHOW STATUS
[ERROR] 16:07:04.061 [ShardingSphere-Command-0] o.a.s.s.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
```

### Example codes for reproduce this issue (such as a github link).

Here is my log from a43eda710d353f7c34b6423e423b83f978dec4be :
```
[INFO ] 17:03:55.938 [ShardingSphere-Command-2] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:03:55.938 [ShardingSphere-Command-2] ShardingSphere-SQL - Logic SQL: SET PROFILING = 1
[INFO ] 17:03:55.938 [ShardingSphere-Command-2] ShardingSphere-SQL - SQLStatement: SetStatement(super=DALStatement(super=AbstractSQLStatement(sqlSegments=[], tables=Tables(tables=[]), logicSQL=SET PROFILING = 1, parametersCount=0)))
[INFO ] 17:03:55.939 [ShardingSphere-Command-2] ShardingSphere-SQL - Actual SQL: cloud_0 ::: SET PROFILING = 1
[INFO ] 17:03:55.942 [ShardingSphere-Command-3] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:03:55.942 [ShardingSphere-Command-3] ShardingSphere-SQL - Logic SQL: SHOW STATUS
[INFO ] 17:03:55.942 [ShardingSphere-Command-3] ShardingSphere-SQL - SQLStatement: ShowOtherStatement(super=DALStatement(super=AbstractSQLStatement(sqlSegments=[], tables=Tables(tables=[]), logicSQL=SHOW STATUS, parametersCount=0)))
[INFO ] 17:03:55.942 [ShardingSphere-Command-3] ShardingSphere-SQL - Actual SQL: cloud_0 ::: SHOW STATUS
[INFO ] 17:03:55.954 [ShardingSphere-Command-4] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:03:55.955 [ShardingSphere-Command-4] ShardingSphere-SQL - Logic SQL: SHOW STATUS
[INFO ] 17:03:55.955 [ShardingSphere-Command-4] ShardingSphere-SQL - SQLStatement: ShowOtherStatement(super=DALStatement(super=AbstractSQLStatement(sqlSegments=[], tables=Tables(tables=[]), logicSQL=SHOW STATUS, parametersCount=0)))
[INFO ] 17:03:55.955 [ShardingSphere-Command-4] ShardingSphere-SQL - Actual SQL: cloud_0 ::: SHOW STATUS
[ERROR] 17:03:55.968 [ShardingSphere-Command-5] o.a.s.s.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
        at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
        at java.util.concurrent.ConcurrentHashMap.containsKey(ConcurrentHashMap.java:964)
        at java.util.concurrent.ConcurrentHashMap$KeySetView.contains(ConcurrentHashMap.java:4558)
        at org.apache.shardingsphere.shardingproxy.backend.schema.Logic
```

`[INFO ] 17:55:33.659 [ShardingSphere-Command-1] ShardingSphere-SQL - Actual SQL: cloud_1 ::: /* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation
[INFO ] 17:55:36.664 [ShardingSphere-Command-7] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:55:36.664 [ShardingSphere-Command-7] ShardingSphere-SQL - Logic SQL: /* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation
[INFO ] 17:55:36.665 [ShardingSphere-Command-7] ShardingSphere-SQL - SQLStatement: SelectStatement(super=DMLStatement(super=AbstractSQLStatement(sqlSegments=[org.apache.shardingsphere.core.parse.sql.segment.dml.SelectItemsSegment@744e7a7d, org.apache.shardingsphere.core.parse.sql.segment.dml.WhereSegment@3ea2a174, org.apache.shardingsphere.core.parse.sql.segment.dml.predicate.SubqueryPredicateSegment@429fb7a], tables=Tables(tables=[]), logicSQL=/* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation, parametersCount=0), shardingConditions=Conditions(orConditions=[]), encryptConditions=Conditions(orConditions=[])), items=[CommonSelectItem(expression=@@session.tx_isolation, alias=Optional.absent())], groupByItems=[], orderByItems=[], toAppendOrderByItems=false, containStar=false, containsSubquery=false, firstSelectItemStartIndex=0, selectListStopIndex=68, groupByLastIndex=0, offset=null, rowCount=null, subqueryStatement=null, subqueryShardingConditions=[])
[INFO ] 17:55:36.665 [ShardingSphere-Command-7] ShardingSphere-SQL - Actual SQL: cloud_0 ::: /* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation
[INFO ] 17:55:38.718 [ShardingSphere-Command-3] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:55:38.718 [ShardingSphere-Command-3] ShardingSphere-SQL - Logic SQL: /* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation
[INFO ] 17:55:38.718 [ShardingSphere-Command-3] ShardingSphere-SQL - SQLStatement: SelectStatement(super=DMLStatement(super=AbstractSQLStatement(sqlSegments=[org.apache.shardingsphere.core.parse.sql.segment.dml.SelectItemsSegment@51fc25f8, org.apache.shardingsphere.core.parse.sql.segment.dml.WhereSegment@c883e0c, org.apache.shardingsphere.core.parse.sql.segment.dml.predicate.SubqueryPredicateSegment@741508c5], tables=Tables(tables=[]), logicSQL=/* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation, parametersCount=0), shardingConditions=Conditions(orConditions=[]), encryptConditions=Conditions(orConditions=[])), items=[CommonSelectItem(expression=@@session.tx_isolation, alias=Optional.absent())], groupByItems=[], orderByItems=[], toAppendOrderByItems=false, containStar=false, containsSubquery=false, firstSelectItemStartIndex=0, selectListStopIndex=68, groupByLastIndex=0, offset=null, rowCount=null, subqueryStatement=null, subqueryShardingConditions=[])
[INFO ] 17:55:38.718 [ShardingSphere-Command-3] ShardingSphere-SQL - Actual SQL: cloud_0 ::: /* ApplicationName=DataGrip 2019.1.2 */ SELECT @@session.tx_isolation
[INFO ] 17:55:38.728 [ShardingSphere-Command-4] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:55:38.728 [ShardingSphere-Command-4] ShardingSphere-SQL - Logic SQL: /* ApplicationName=DataGrip 2019.1.2 */ SHOW WARNINGS
[INFO ] 17:55:38.728 [ShardingSphere-Command-4] ShardingSphere-SQL - SQLStatement: ShowOtherStatement(super=DALStatement(super=AbstractSQLStatement(sqlSegments=[], tables=Tables(tables=[]), logicSQL=/* ApplicationName=DataGrip 2019.1.2 */ SHOW WARNINGS, parametersCount=0)))
[INFO ] 17:55:38.728 [ShardingSphere-Command-4] ShardingSphere-SQL - Actual SQL: cloud_0 ::: /* ApplicationName=DataGrip 2019.1.2 */ SHOW WARNINGS
[INFO ] 17:55:38.731 [ShardingSphere-Command-5] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:55:38.732 [ShardingSphere-Command-5] ShardingSphere-SQL - Logic SQL: /* ApplicationName=DataGrip 2019.1.2 */ SET net_write_timeout=600
[INFO ] 17:55:38.732 [ShardingSphere-Command-5] ShardingSphere-SQL - SQLStatement: SetStatement(super=DALStatement(super=AbstractSQLStatement(sqlSegments=[], tables=Tables(tables=[]), logicSQL=/* ApplicationName=DataGrip 2019.1.2 */ SET net_write_timeout=600, parametersCount=0)))
[INFO ] 17:55:38.732 [ShardingSphere-Command-5] ShardingSphere-SQL - Actual SQL: cloud_0 ::: /* ApplicationName=DataGrip 2019.1.2 */ SET net_write_timeout=600
[ERROR] 17:55:38.733 [ShardingSphere-Command-6] o.a.s.s.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
        at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
        at java.util.concurrent.ConcurrentHashMap.containsKey(ConcurrentHashMap.java:964)
        at java.util.concurrent.ConcurrentHashMap$KeySetView.contains(ConcurrentHashMap.java:4558)
        at org.apache.shardingsphere.shardingproxy.backend.schema.LogicSchemas.schemaExists(LogicSchemas.java:138)
        at org.apache.shardingsphere.shardingproxy.backend.text.admin.UseDatabaseBackendHandler.execute(UseDatabaseBackendHandler.java:51)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:72)
        at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:86)
        at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:66)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[INFO ] 17:55:38.736 [ShardingSphere-Command-7] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 17:55:38.736 [ShardingSphere-Command-7] ShardingSphere-SQL - Logic SQL: /* ApplicationName=DataGrip 2019.1.2 */ SHOW WARNINGS
[INFO ] 17:55:38.736 [ShardingSphere-Command-7] ShardingSphere-SQL - SQLStatement: ShowOtherStatement(super=DALStatement(super=AbstractSQLStatement(sqlSegments=[], tables=Tables(tables=[]), logicSQL=/* ApplicationName=DataGrip 2019.1.2 */ SHOW WARNINGS, parametersCount=0)))
[INFO ] 17:55:38.736 [ShardingSphere-Command-7] ShardingSphere-SQL - Actual SQL: cloud_0 ::: /* ApplicationName=DataGrip 2019.1.2 */ SHOW WARNINGS
[INFO ] 17:55:38.740 [ShardingSphere-Command-0] ShardingSphere-SQL - Rule Type: sharding`
This is my current log

--- a/sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/resources/META-INF/parsing-rule-definition/encrypt/mysql/sql-statement-rule-definition.xml
+++ b/sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/resources/META-INF/parsing-rule-definition/encrypt/mysql/sql-statement-rule-definition.xml
@@ -48,7 +48,7 @@
     <sql-statement-rule context="setRole" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dcl.DCLStatement" />
     <sql-statement-rule context="setPassword" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dcl.DCLStatement" />
     
-    <sql-statement-rule context="use" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.UseStatement" />
+    <sql-statement-rule context="use" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.UseStatement" extractor-rule-refs="schema" />
     <sql-statement-rule context="desc" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.DescribeStatement" />
     <sql-statement-rule context="showDatabases" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.ShowDatabasesStatement" />
     <sql-statement-rule context="showTables" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.ShowTablesStatement" extractor-rule-refs="fromSchema" />

--- a/sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/resources/META-INF/parsing-rule-definition/masterslave/mysql/sql-statement-rule-definition.xml
+++ b/sharding-core/sharding-core-parse/sharding-core-parse-mysql/src/main/resources/META-INF/parsing-rule-definition/masterslave/mysql/sql-statement-rule-definition.xml
@@ -48,7 +48,7 @@
     <sql-statement-rule context="setRole" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dcl.DCLStatement" />
     <sql-statement-rule context="setPassword" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dcl.DCLStatement" />
     
-    <sql-statement-rule context="use" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.UseStatement" />
+    <sql-statement-rule context="use" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.UseStatement" extractor-rule-refs="schema" />
     <sql-statement-rule context="desc" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.DescribeStatement" extractor-rule-refs="table" />
     <sql-statement-rule context="showDatabases" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.ShowDatabasesStatement" />
     <sql-statement-rule context="showTables" sql-statement-class="org.apache.shardingsphere.core.parse.sql.statement.dal.dialect.mysql.ShowTablesStatement" extractor-rule-refs="fromSchema" />

